core/unit_test/TestViewMapping_a.hpp

```diff
@@ -653,64 +653,66 @@ void test_view_mapping() {
-    ASSERT_EQ(cr1.span(), size_t(N));
-    ASSERT_EQ(vr1.data(), &data[0]);
-    ASSERT_EQ(cr1.data(), &data[0]);
+    static_assert(
+        std::is_same_v<typename T::memory_space, typename Space::memory_space>);
+    static_assert(std::is_same_v<typename T::reference_type, int&>);
 
-    ASSERT_TRUE((std::is_same_v<typename T::data_type, int*>));
-    ASSERT_TRUE((std::is_same_v<typename T::const_data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename T::non_const_data_type, int*>));
+    static_assert(T::rank == size_t(1));
 
-    ASSERT_TRUE((std::is_same_v<typename T::scalar_array_type, int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename T::const_scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename T::non_const_scalar_array_type, int*>));
+    static_assert(std::is_same_v<typename C::data_type, const int*>);
+    static_assert(std::is_same_v<typename C::const_data_type, const int*>);
+    static_assert(std::is_same_v<typename C::non_const_data_type, int*>);
 
-    ASSERT_TRUE((std::is_same_v<typename T::value_type, int>));
-    ASSERT_TRUE((std::is_same_v<typename T::const_value_type, const int>));
-    ASSERT_TRUE((std::is_same_v<typename T::non_const_value_type, int>));
+    static_assert(std::is_same_v<typename C::scalar_array_type, const int*>);
+    static_assert(
+        std::is_same_v<typename C::const_scalar_array_type, const int*>);
+    static_assert(
+        std::is_same_v<typename C::non_const_scalar_array_type, int*>);
 
-    ASSERT_TRUE((std::is_same_v<typename T::memory_space,
-                                typename Space::memory_space>));
-    ASSERT_TRUE((std::is_same_v<typename T::reference_type, int&>));
+    static_assert(std::is_same_v<typename C::value_type, const int>);
+    static_assert(std::is_same_v<typename C::const_value_type, const int>);
+    static_assert(std::is_same_v<typename C::non_const_value_type, int>);
 
-    ASSERT_EQ(T::rank, size_t(1));
+    static_assert(
+        std::is_same_v<typename C::memory_space, typename Space::memory_space>);
+    static_assert(std::is_same_v<typename C::reference_type, const int&>);
 
-    ASSERT_TRUE((std::is_same_v<typename C::data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename C::const_data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename C::non_const_data_type, int*>));
+    static_assert(C::rank == size_t(1));
 
-    ASSERT_TRUE((std::is_same_v<typename C::scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename C::const_scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename C::non_const_scalar_array_type, int*>));
+    if (Kokkos::SpaceAccessibility<Kokkos::HostSpace,
+                                   typename Space::memory_space>::accessible &&
+        Kokkos::SpaceAccessibility<typename Space::memory_space,
+                                   Kokkos::HostSpace>::assignable) {
```

Comment:
This is a bit confusing.  Should that be a skip?

Reply:
A skip implies `return` and there is more code in this test that is executed before and after so the test is still meaningful when the View is not host-accessible (but just tests less). 

We could simplify the condition to `std::is_same_v<Space::memory_space, Kokkos::HostSpace>` which would likely have the same effect as of now but would also skip the test for the HMM-like architectures going forward that could execute it.